File: app/src/main/java/com/juawapps/newsspread/data/api/adapters/WrappedApiResponse.java
package com.juawapps.newsspread.data.api.adapters;

import com.google.gson.annotations.SerializedName;

/**
 * Default object response from the api.
 */
@SuppressWarnings({"unused", "WeakerAccess"})
class WrappedApiResponse<Object> {

    private String status;
    @SerializedName(value="articles", alternate={"sources"})
    private
    Object content;

    public String getStatus() {
        return status;
    }

    public Object getContent() {
        return content;
    }
}

File: app/src/main/java/com/juawapps/newsspread/data/api/adapters/WrappedApiResponse.kt
package com.juawapps.newsspread.data.api.adapters

import com.google.gson.annotations.SerializedName

/**
 * Default object response from the api.
 */
internal class WrappedApiResponse<T> {

    var status: String? = null
    @SerializedName(value = "articles", alternate = ["sources"])
    var content: T? = null
}
